flexget/plugins/input/imdb_watchlist.py

```diff
@@ -84,72 +85,145 @@ def on_task_input(self, task, config):
+            entries = self.parse_react_widget(task, config, url, params, headers)
+        else:
+            entries = self.parse_html_list(task, config, url, params, headers)
+        return entries
 
-        headers = {'Accept-Language': config.get('force_language')}
+    def fetch_page(self, task, url, params, headers):
         log.debug('Requesting: %s %s', url, headers)
-
         try:
             page = task.requests.get(url, params=params, headers=headers)
-        except HTTPError as e:
-            raise plugin.PluginError(e.args[0])
+        except RequestException as e:
+            raise plugin.PluginError(str(e))
         if page.status_code != 200:
-            raise plugin.PluginError('Unable to get imdb list. Either list is private or does not exist.')
+            raise plugin.PluginError(
+                'Unable to get imdb list. Either list is private or does not exist.' +
+                ' Html status code was: %d.' % page.status_code)
+        return page
 
-        soup = get_soup(page.text)
+    def parse_react_widget(self, task, config, url, params, headers):
+        page = self.fetch_page(task, url, params, headers)
+        try:
+            re_result = re.search('IMDbReactInitialState.push\((.+?)\);\n', page.text)
+            json_vars = json.loads(re_result.group(1))
+        except (TypeError, AttributeError, ValueError) as e:
+            raise plugin.PluginError(
+                'Unable to get imdb list from imdb react widget.' +
+                ' Either the list is empty or the imdb parser of the imdb_watchlist plugin is broken.' +
+                ' Original error: %s.' % str(e))
+        total_item_count = 0
+        if 'list' in json_vars and 'items' in json_vars['list']:
+            total_item_count = len(json_vars['list']['items'])
+        if not total_item_count:
+            log.verbose('No movies were found in imdb list: %s', config['list'])
+            return
+        imdb_ids = []
+        for item in json_vars['list']['items']:
+            if is_valid_imdb_title_id(item.get('const')):
+                imdb_ids.append(item['const'])
+        params = {'ids': ','.join(imdb_ids)}
+        url = 'http://www.imdb.com/title/data'
+        page = self.fetch_page(task, url, params, headers)
+        try:
+            json_data = json.loads(page.text)
+        except (ValueError, TypeError) as e:
+            raise plugin.PluginError('Unable to get imdb list from imdb JSON API.' +
+                ' Either the list is empty or the imdb parser of the imdb_watchlist plugin is broken.' +
+                ' Original error: %s.' % str(e))
+        log.verbose('imdb list contains %d items', len(json_data))
+        log.debug('First entry (imdb id: %s) looks like this: %s', imdb_ids[0], json_data[imdb_ids[0]])
+        entries = []
+        for imdb_id in imdb_ids:
+            imdb_type = ''
+            entry = Entry()
+            if not ('title' in json_data[imdb_id] and
```

Comment:
Redundant parentheses

Reply:
I deliberately added the parentheses here in order to be able to break the long if statement into several lines (as suggested by cvium). 
Also: AFAIK, I would have to change this to `if not ... or not ... or not ... :` in order to remove the parentheses and still get the same result. Therefore, the parentheses save me several additional negations.
For these reasons, I would like to keep the parentheses if that is o.k.

Reply:
Sure